python-sdk/src/astro/dataframes/pandas.py

```diff
@@ -20,7 +20,7 @@ def serialize(self):
         df_size = self.memory_usage(deep=True).sum()
         if df_size < (settings.MAX_DATAFRAME_MEMORY_FOR_XCOM_DB * 1024):
             logger.info("Dataframe size: %s bytes. Storing it in Airflow's metadata DB", df_size)
-            return self.to_json()
+            return self.to_dict()
```

Comment:
Should we add test for this change? coverage is complaining

Reply:
https://github.com/astronomer/astro-sdk/blob/6e0b9eee84b96759ae0249575034aeb1a340edc5/python-sdk/tests/dataframes/pandas.py#L69-L73